test/test_core.py

```diff
@@ -375,7 +375,13 @@ def decorator(f):
 
       @wraps(f)
       def decorated(self, *args, **kwargs):
-        if (name + '=1') in self.cflags or self.get_setting(name):
+        nonlocal name
+        if '=' in name:
+          key, val = name.split('=', 1)
+        else:
+          key = name
+          val = 1
+        if (f'{key}={val}') in self.cflags or self.get_setting(key) == int(val):
```

Comment:
We might want to handle both `-sFOO=1` and just `-sFOO`.

However, `-sFOO` should be far more common, if not universal.

Reply:
> We might want to handle both `-sFOO=1` and just `-sFOO`.

The old code only handled `name + '=1'`, so it didn't handle `-sFOO` either.

Now that I read it, it looks to me that because `name` is always of form `FOO`, the old code doesn't look like it matched anything, since 'FOO=1` would never appear in `self.cflags`. (it would be `-sFOO=1` or `-sFOO` there, not `FOO=1`)

Updated to handle both.